### PR TITLE
Upload initial version of NHibernate.MySqlConnector

### PR DESCRIPTION
**Pull request by razzmatazz**
This is in response to https://github.com/nhibernate/nhibernate-core/issues/1375

There are a couple of things that I am not sure what to do about:
 - how to build a nuget package
 - should we provide support for multiple frameworks in .csproj (as it uses .netstandard2.0 now)
 - what version of NHibernate should I reference in .csproj (now it has 5.2.1 specified)

### CODE REVIEW

File: src/Driver/MySqlConnectorDriver.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		public override bool SupportsMultipleQueries => true;
+
+		public override bool RequiresTimeSpanForTime => true;
+
+		// As of v5.7, lower dates may "work" but without guarantees.
+		// https://dev.mysql.com/doc/refman/5.7/en/datetime.html
+		/// <inheritdoc />
+		public override DateTime MinDate => new DateTime(1000, 1, 1);
+
+		System.Type IEmbeddedBatcherFactoryProvider.BatcherFactoryClass => typeof(MySqlClientBatchingBatcherFactory);
```
Comment (danspam) [inline src/Driver/MySqlConnectorDriver.cs]: Using `MySqlClientBatchingBatcherFactory` does not work when using MySqlConnector. `MySqlClientBatchingBatcher` creates a `MySqlClientSqlCommandSet` which uses reflection on the `MySql.Data` assembly to create a `MySqlDataAdapter` and that causes an exception.

All three of those classes will need to be recreated to use the MySqlConnector MySqlDataAdapter instead.

**Comment by razzmatazz** [comment]
@danspam that is easily out of my competence level..

File: src/Driver/MySqlConnectorDriver.cs (hunk 1 above)
Comment (dennis-gr) [reply src/Driver/MySqlConnectorDriver.cs]: AFAICT there's no need for a specialized batcher. Can just use `GenericBatchingBatcherFactory` here.

File: src/Driver/MySqlConnectorDriver.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		/// <summary>
+		/// MySql.Data does not support preparing of commands.
+		/// </summary>
+		/// <value><see langword="false" /> - it is not supported.</value>
+		/// <remarks>
+		/// With the Gamma MySql.Data provider it is throwing an exception with the 
+		/// message "Expected End of data packet" when a select command is prepared.
+		/// </remarks>
+		protected override bool SupportsPreparingCommands => false;
+
```
Comment (dennis-gr) [inline src/Driver/MySqlConnectorDriver.cs]: MySqlConnector does support preparing commands.

Comment (danspam) [reply src/Driver/MySqlConnectorDriver.cs]: Not sure it does yet? https://github.com/mysql-net/MySqlConnector/issues/397

File: src/Driver/MySqlConnectorDriver.cs (hunk 1 above)
Comment (danspam) [reply src/Driver/MySqlConnectorDriver.cs]: It does seem that way - [I asked the question](https://groups.google.com/d/msgid/nhibernate-development/413aa7e6-8445-40a0-90ef-a4f2c444adb9%40googlegroups.com.) on the discussion group for NHibernate and it seems the generic batcher is a more recent addition and appears to work find with MySql.

File: src/Driver/MySqlConnectorDriver.cs (hunk 2 above)
Comment (dennis-gr) [reply src/Driver/MySqlConnectorDriver.cs]: It was implemented in 0.43, see [here](https://github.com/mysql-net/MySqlConnector/pull/534)

**Comment by jemiller0** [comment]
Any word on whether support for MySqlConnector will be available any time soon? I want to get rid of Oracle's garbage driver.

**Comment by hazzik** [comment]
Thanks, I've chosen #2 as it has all issues mentioned here resolved.

**Pull request closed without merging**